dts/arm/silabs/efm32pg12b.dtsi

```diff
@@ -182,6 +182,17 @@
 				#gpio-cells = <2>;
 			};
 		};
+
+		adc0: adc@40002000 {
+			compatible = "silabs,gecko-adc";
+			reg = <0x40002000 0x400>;
+			interrupts = <15 0>;
+			peripheral-id = <0>;
+			prescaler = <32>;
+			label = "ADC_0";
+			status = "disabled";
+			#io-channel-cells = <1>;
+		};
```

Comment:
Could you also add the following code
```
		adc0: adc@40002000 {
			compatible = "silabs,gecko-adc";
			reg = <0x40002000 0x400>;
			interrupts = <14 0>;
			peripheral-id = <0>;
			prescaler = <32>;
			label = "ADC_0";
			status = "disabled";
			#io-channel-cells = <1>;
		};
```
to `dts/arm/silabs/efr32fg1p.dtsi`. I'm using efr32_slwstk6061a board for testing. All DTS related changes can be submitted in one commit, maybe even squashed with the commit introducing the driver code since they can't exists separately.

Reply:
As you requested, added support for the Flex Gecko. I have also a Giant Gecko board where I'm testing on, but don't have access to any of the other boards, so your help in testing is appreciated!